terminology_bot/bot.py

```diff
@@ -89,44 +83,47 @@ def update_state_handlers(self):
                          self.choose_menu_option, pass_user_data=True)
         )
 
-    def start(self, bot, update):
+    def start(self, bot, update, user_data):
         """
         Sends the greeting message with the start menu: 'Add new term' and 'Get list of terms' options
         :return: the state START_MENU
         """
         # set language settings, UI-elements and button handlers according User's language_code
         lang_code = update.message.from_user.language_code
-        if lang_code != 'en':
-            change_language(lang_code)
-
-        self.set_keyboard()
-        self.update_state_handlers()
+        print(update.message.from_user.first_name, lang_code)
+        lang, options, start_btn, term_btn = self.set_language_and_options(lang_code)
+        user_data.update({'lang': lang, 'options': options, 'start_btn': start_btn, 'term_btn': term_btn})
 
+        _ = user_data['lang'].gettext
```

Comment:
This line is similar in all method.
Is it ok?

Reply:
This is a way to define user specific translation function. Do you know how to make it another way?